### ISSUE TITLE
We NEED Google (and optionally Apple/ Facebook) Authentication via Firebase

### ISSUE THREAD

**Issue by easaw** [thunkable/thunkable-issues#529]
Tens of thousands of apps and websites allow Google Authentication as a convenient and safe login. Opening up the Thunk Firebase Auth service to accommodate would be a huge advantage to just email, which is almost useless in this day and age.

I would like Google (most important), Apple or even Facebook Auth to be added to the Firebase Authentication options.

This is a crucial feature that I believe absolutely Thunk should have, because users don't want to create email logins any more. We are not keeping up with the modern conveniences. Plus Google (and Apple) Auth options via Firebase make apps more secure, something developers need to provide reliable, trustworthy services for their users. Thunkable can help here by making this a reality. 

Thunkable ✕ (x.thunkable.com)

I mean, the Thunk Live app has Goolgle login, so clearly the Thunk Dev Team already has experience implementing this! 😊

**Comment by Bellacopia** [comment]
I fully agree: logging in with Google, Apple or Facebook is now almost essential.

**Comment by domhnallohanlon** [comment]
Hey @easaw and @Bellacopia,

We don't have a component for this, but there is a post in the Community that outlines how to do something like this manually:
https://community.thunkable.com/t/facebook-and-google-signup-login-in-your-app/841096?u=domhnallohanlon

**Comment by MrMazzone** [comment]
My issue is this:  I have implemented ways for Google login to launch via the default web browser and then user navigates back to app and is logged in.  However, Apple will not accept this and login cannot be completed in web viewer as it is blocked.  Apple's app guidelines state the following and all apps like how I have implemented sign in will not pass. 
![image](https://user-images.githubusercontent.com/40954645/115043615-b005e900-9ea2-11eb-800f-95599b4d9d65.png)

This is why we need a Thunkable component for either:
1. Google/Apple/FB Sign In 
2. or Chrome Custom Tabs on Android and SFSafariViewController on iOS (#598)

This is very important for creating apps with user authentication.
